web/docs/tql2/operators/from.md

```diff
@@ -3,3 +3,241 @@ sidebar_class_name: hidden
+#### Example with pipeline argument:
+
+```tql title="from operator"
+from "tcp://0.0.0.0:12345", parallel=10 {
+  read_gelf
+}
+```
+```tql title="Effective pipeline"
+load_tcp "tcp://0.0.0.0:12345", parallel=10 {
+  read_gelf
+}
+```
+
+## Supported Deductions
+
+### URI schemes
+
+| Scheme | Operator | Example |
+|:------ |:-------- |:------- |
+| `abfs`,`abfss` | [`load_azure_blob_storage`](load_azure_blob_storage.md) | `from "abfs://path/to/file.json"` |
+| `amqp` | [`load_amqp`](load_amqp.md) | `from "amqp://…` |
+| `file` | [`load_file`](load_file.md) | `from "file://path/to/file.json"` |
+| `fluentbit` | [`from_fluent_bit`](from_fluent_bit.md) | `from "fluentbit://elasticsearch"` |
+| `ftp`, `ftps` | [`load_ftp`](load_ftp.md) | `from "ftp://example.com/file.json"` |
+| `gcps` | [`load_google_cloud_pubsub`](load_google_cloud_pubsub.md) | `from "gcps://project_id/subscription_id"` |
+| `http`, `https` | [`load_http`](load_http.md) | `from "http://example.com/file.json"` |
+| `kafka` | [`load_kafka`](load_kafka.md) | `from "kafka://topic"` |
+| `s3` | [`load_s3`](load_s3.md) | `from "s3://bucket/file.json"` |
+| `tcp` | [`load_tcp`](load_tcp.md) | `from "tcp://127.0.0.1:13245" { read_json }` |
+| `udp` | [`load_udp`](load_udp.md) | `from "udp://127.0.0.1:56789"` |
+
+Please see the respective operator pages for details on the URI's locator format.
+
+### File extensions
+
+#### Format
+
+The `from` operator can deduce the file format based on these file-endings:
+
+| Format | File Endings | Operator  |
+|:------ |:------------ |:--------- |
+|  CSV  | `.csv` | [`read_csv`](read_csv.md) |
+|  CEF  | `.cef` | [`read_cef`](read_cef.md) |
+|  Feather  | `.feather`, `.arrow` | [`read_feather`](read_feather.md) |
+|  JSON  | `.json` | [`read_json`](read_json.md) |
+|  LEEF  | `.leef` | [`read_leef`](read_leef.md) |
+|  NDJSON  | `.ndjson`, `.jsonl` | [`read_ndjson`](read_ndjson.md) |
+|  Parquet  | `.parquet` | [`read_parquet`](read_parquet.md) |
+|  Pcap  | `.pcap` | [`read_pcap`](read_pcap.md) |
+|  SSV  | `.ssv` | [`read_ssv`](read_ssv.md) |
+|  TSV  | `.tsv` | [`read_tsv`](read_tsv.md) |
+|  YAML  | `.yaml` | [`read_yaml`](read_yaml.md) |
+
+#### Compression
+
+The `from` operator can deduce the following compressions based on these
+file-endings:
+
+| Compression |    File Endings  |
+|:----------- |:---------------- |
+| Brotli      | `.br`, `.brotli` |
+| Bzip2       | `.bz2`           |
+| Gzip        | `.gz`, `.gzip`   |
+| LZ4         | `.lz4`           |
+| Zstd        | `.zst`, `.zstd`  |
+
+## Examples
+
+### Load a local file
+
+```tql
+from "path/to/my/load/file.csv"
+```
+
+### Load a compressed file
+
+```tql
+from "path/to/my/load/file.json.bz2"
+```
+
+### Load a file with parser arguments
+
+Note how the explicit `decompress` step is now necessary:
+
+```tql
+from "path/to/my/load/file.json.bz2" {
+  decompress "brotoli" // this is now necessary due to the pipeline argument
+  read_ndjson selector="event_type:suricata"
+}
+```
```

Comment:
This example needs some additional explanation, e.g., that it would normally use `read_json`, but that we want `read_suricata` or `read_ndjson` with some arguments (Suricata is a bad example for the latter because of the dedicated parser).